packages/plugin-data/src/api/data/soql/reporters.ts

```diff
@@ -0,0 +1,326 @@
+    }
+
+    return { attributeNames, children, aggregates };
+  }
+
+  /* eslint-disable @typescript-eslint/no-explicit-any */
+  public soqlQuery(columns: string[], records: any[], totalCount: number): void {
+    this.prepNullValues(records);
+    this.ux.table(records, { columns: this.prepColumns(columns) });
+    this.log(chalk.bold(messages.getMessage('displayQueryRecordsRetrieved', [totalCount])));
+  }
+
+  /* eslint-disable @typescript-eslint/no-explicit-any */
+  public prepNullValues(records: any[]): void {
+    records.forEach((record): void => {
+      Reflect.ownKeys(record).forEach((propertyKey) => {
+        const value = Reflect.get(record, propertyKey);
+        if (value === null) {
+          Reflect.set(record, propertyKey, chalk.bold('null'));
+        } else if (typeof value === 'object') {
+          this.prepNullValues([value]);
+        }
+      });
+    });
+  }
+
+  public prepColumns(columns: string[]): ColumnAttributes[] {
+    return columns.map(
+      (field: string): ColumnAttributes => ({
+        key: field,
+        label: field.toUpperCase(),
+      })
+    );
+  }
+
+  /* eslint-disable @typescript-eslint/no-explicit-any */
+  public massageRows(queryResults: any[], children: any[], aggregates: FunctionField[]): any {
+    // There are subqueries or aggregates. Massage the data.
+    if (children.length > 0 || aggregates.length > 0) {
+      queryResults = queryResults.reduce((newResults, result) => {
+        newResults.push(result);
+
+        // Aggregates are soql functions that aggregate data, like "SELECT avg(total)" and
+        // are returned in the data as exprX. Aggregates can have aliases, like "avg(total) totalAverage"
+        // and are returned in the data as the alias.
+        if (aggregates.length > 0) {
+          for (let i = 0; i < aggregates.length; i++) {
+            const aggregate = aggregates[i];
+            if (!aggregate.alias) {
+              Reflect.set(result, aggregate.name, Reflect.get(result, `expr${i}`));
+            }
+          }
+        }
+
+        if (children.length > 0) {
+          const childrenRows = Object.assign({});
+          children.forEach((child) => {
+            Reflect.set(childrenRows, child, Reflect.get(result, child));
+            Reflect.deleteProperty(result, child);
+          });
+
+          Reflect.ownKeys(childrenRows).forEach((child) => {
+            if (childrenRows[child]) {
+              childrenRows[child].records.forEach((record: any) => {
+                const newRecord = Object.assign({});
+                Object.entries(record).forEach(([key, value]) => {
+                  Reflect.defineProperty(newRecord, `${child.toString()}.${key}`, { value });
+                });
+                newResults.push(newRecord);
+              });
+            }
+          });
+        }
+        return newResults;
+      }, []);
+    }
+    return queryResults;
+  }
+}
+// /* eslint-enable @typescript-eslint/no-explicit-any */
```

Comment:
what any is this rule ignoring? I _think_ it might be extra?

Reply:
@WillieRuemmele, extra it is. Thanks!